### PR TITLE
Add retry logic to Anidex

### PR DESCRIPTION
**Pull request by 6cUbi57z**
Anidex seems to have been having quite a lot of issues ongoing for a few months now. The pages are returning 500 errors but if you refresh a few times the page loads. Obviously the ideal solution would be for Anidex to resolve this but not much seems to be happening there and this is causing the indexer to go offline in Sonarr.

This change adds retry logic using [Polly](https://github.com/App-vNext/Polly). The retries are configured with a delay between which increases exponentially in order to reduce strain on the server. It will also log a warning message when an error response is received so that the attempts can be tracked.

The default has been set not to retry as the delay will increase the API response times.

Apologies if this isn't meeting coding standards. My VS install is broken so I'm having to use VSCode.

### CODE REVIEW

**Comment by 6cUbi57z** [comment]
Updated following discussion in #11221 .

- The duplicated retry logic has now been removed. Most of the changes in `Anidex` have been rolled back.
- The retry logic in `BaseWebIndexer` (existing previous to this PR) have now been modified to use Polly.
  - The default is 3 retry attempts (same as previous) with the delay starting at 0.5s and increasing exponentially. This is different to previous (always 0.5s), however, there was an additional wait before (there was a delay in the last loop after the last exception). This means that the time to failure should be the same.

I have also added some properties and methods to the `BaseWebIndexer` to simplify the retries per indexer. To change the number of retries for a particular indexer in code, you can override the `NumberOfRetryAttempts` property:

```csharp
protected override int NumberOfRetryAttempts => 5;
```

Alternatively, a user-configurable setting can be added to an indexer by calling the `EnableConfigurableRetryAttempts();` method from the constructor of an indexer. The default can be changed by overriding the `DefaultNumberOfRetryAttempts` property.

This keeps the UI for retries consistent and reduces code duplication, however, it restricts indexers from using the `retryAttempts` config property as it is used by the base class. It may be possible to add some protection against this, or it can be removed if we think configuration is best handled completely in the subclasses.

Two indexers with known stability issues have had user configurable retries enabled: Anidex and RARBG (#11221).

File: src/Jackett.Common/Indexers/Anidex.cs [hunk 1]
```diff
@@ -14,9 +14,11 @@
 using Jackett.Common.Models.IndexerConfig;
 using Jackett.Common.Services.Interfaces;
 using Jackett.Common.Utils;
+using Jackett.Common.Utils.Clients;
```
Comment (garfield69) [inline src/Jackett.Common/Indexers/Anidex.cs:17]: my vs2019 tells me that `using Jackett.Common.Utils.Clients;` and `using Polly;` are redundant

File: src/Jackett.Common/Indexers/BaseIndexer.cs [hunk 2]
```diff
@@ -9,6 +9,8 @@
 using Jackett.Common.Services.Interfaces;
 using Jackett.Common.Utils;
 using Jackett.Common.Utils.Clients;
+using Polly;
+using Polly.Retry;
```
Comment (garfield69) [inline src/Jackett.Common/Indexers/BaseIndexer.cs:13]: my vs2019 tells me `using Polly.Retry;` is redundant

**Review by garfield69 — CHANGES_REQUESTED** [review]
Tested OK.
Mind you, as is usually the case for me, when I tested both RARBG and Anidex I could not get them to fail so I don't get to see the retry in action ;-D

**Review by garfield69 — APPROVED** [review]

**Comment by 6cUbi57z** [comment]
Thanks @garfield69 !

Have a look at Mockoon (https://mockoon.com/). You may have to tweak some URLs so that the indexers point at localhost rather than the actual sites but it should let you force failure responses. Also handy for development so that you don't have to keep hitting the sites.

**Comment by ilike2burnthing** [comment]
[v0.17.710](https://github.com/Jackett/Jackett/releases/tag/v0.17.710)

**Pull request merged**